PySDM/formulae.py

```diff
@@ -40,7 +40,25 @@ def __init__(  # pylint: disable=too-many-locals
         hydrostatics: str = "Default",
         freezing_temperature_spectrum: str = "Null",
         heterogeneous_ice_nucleation_rate: str = "Null",
+        fragmentation_function: str = "AlwaysN",
     ):
+        self.condensation_coordinate = condensation_coordinate
+        self.saturation_vapour_pressure = saturation_vapour_pressure
+        self.hygroscopicity = hygroscopicity
+        self.drop_growth = drop_growth
+        self.surface_tension = surface_tension
+        self.diffusion_kinetics = diffusion_kinetics
+        self.latent_heat = latent_heat
+        self.diffusion_thermics = diffusion_thermics
+        self.ventilation = ventilation
+        self.state_variable_triplet = state_variable_triplet
+        self.particle_advection = particle_advection
+        self.hydrostatics = hydrostatics
+        self.freezing_temperature_spectrum = freezing_temperature_spectrum
+        self.heterogeneous_ice_nucleation_rate = heterogeneous_ice_nucleation_rate
+        self.fragmentation_function = fragmentation_function
```

Comment:
I feel that in order to use it here, classes that represent `fragmentation_function ` should have a common API. Moreover, we use it in code in a different way than other members of Formulae.

Reply:
What do you think of creating static method in Formulae that will transform python one-liners into c code?

Reply:
- re common API, so far in all cases of the components of `Formulae` it was the case, here we indeed break it, but is it a problem? (the rationale is exemplified, e.g., with the need for the `sigma1` function in Straub)
- re "we use it in code in a different way than other members of Formulae" - please elaborate
- re "creating static method in Formulae that will transform python one-liners into c code", the issue here is that we need `self` to get values of constants from, and then we also need to know which implementation for a component was selectd when constructing `Formulae` - these are the reasons why `c_inline` is not static

Reply:
Do I understand correctly, that the roles of Formulae are:
- Parametrization (e.g. by providing `saturation_vapour_pressure` param in `Formulae`, we can choose between different approximations of saturation vapour pressure)
- Translation of Python one-liners into C++ code

Reply:
exactly, plus:
- generating constant catalogue by merging default values with possibly overwritten ones passed to `Formulae::__init__`
- passing the constants catalogue as the first arg to all the one-liners
- orchestrating Numba-njitting of all the one-liners
- handling vectorisation (see `__vectorize` in https://github.com/atmos-cloud-sim-uj/PySDM/blob/main/PySDM/formulae.py#L192)
- handling access to helper routines for selected one-liners (see `__extras` in https://github.com/atmos-cloud-sim-uj/PySDM/blob/96935d1576812293fb7b21eebce5a0f967203910/PySDM/formulae.py#L185)

The last two items are exemplified here: https://github.com/atmos-cloud-sim-uj/PySDM/blob/main/PySDM/physics/surface_tension/compressed_film_ruehl.py